docs/src/routes/docs/components/autocomplete-field.svx

```diff
@@ -0,0 +1,133 @@
+
+<RelatedComponents components={relatedComponents} />
+
+<Showcase>
+  <div slot="showcase">
+    <AutocompleteField {getOptions} bind:selection />
+    <pre>{JSON.stringify(selection, null, 2)}</pre>
+  </div>
+  <div slot="source">
+
+```svelte
+<script>
+  async function* getOptions(text) {
+    await sleep(1000);
+    yield [
+      { name: text, details: 'Optional' },
+      { name: `it highlights the match: ${text}` },
+    ];
+  }
+
+  let selection = [];
+</script>
+
+<AutocompleteField {getOptions} bind:selection />
+<pre>{JSON.stringify(selection, null, 2)}</pre>
+```
+
+  </div>
+</Showcase>
+
+
+## Properties {#properties}
+
+### Functional Properties
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| ==**`getOptions`**==&nbsp;<span class="required">*</span> | `async function*` | An async generator of suggestions. <br /> Receives the input from the text field and is expected to yield promises that resolve to arrays of objects `{ name, details }`. |
+| **`selection`** | `Array` | The current selection as an array of objects `{ name, details }`. <br /> Can be used to set the selection programmatically. |
+| **`minSearchLength`** | `number` | The minimum amount of characters to call ==`getOptions`==. <br /> 3 by default. |
+| **`maxOptions`** | `number` | The maximum amount of options than can be selected. Set to `null` (the default) for no restrictions. |
+| **`searchQuery`** | `number` | The current value of the text field. Can be used to control the query programmatically. |
+| **`optionComponent`** | Svelte component | The component used to render a suggestion in the list. <br /> Receives two props: the `option` object (`{ name, details }`) and the current `query`. <br /> Expected to dispatch `click` events on selection. <br /> Defaults to [==`AutocompleteOption`==](./docs/components/autocomplete-option). |
+| **`focus`** | `boolean` | Allows to programmatically control the suggestions dropdown. |
+| **`...`** | `any` | Everything else will be passed to the underlying `<input>` element. |
+
+<div class="required">* Required</div>
+
+### Class Properties
+
+All `<AutocompleteField>` components use the `.autocomplete-field` class.
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| **`class`** | `string` | A class string to add to the component. |
+
+
+## Events {#events}
+
+| Name | Event Detail | Description |
+| ---- | ------------ | ----------- |
+| **`change`** | `{ value }` | Fired when the selection changes. The `value` contains the selection array. |
+
+
+## Slots {#slots}
+
+### `too-many-options` slot
+
+The message to show in the dropdown when the maximum amount of selected options has been reached.
+
+### `not-enough-input` slot
+
+The message to show in the dropdown when the minimum amount of characters to type for searching hasn't been reached.
+
+### `loading-options` slot
+
+The loading state of the dropdown while the next promise with suggestions is resolving.
```

Comment:
How about mentioning the default here? Or is the showcase enough?

Reply:
I think it is